packages/pluggableWidgets/datagrid-web/src/Datagrid.tsx

```diff
@@ -23,6 +26,12 @@ export default function Datagrid(props: DatagridContainerProps): ReactElement {
         }
     });
 
+    useEffect(() => {
+        if (props.datasource.filter && !filtered && !viewStateFilters.current) {
+            viewStateFilters.current = props.datasource.filter;
```

Comment:
So the datagrid has been filtered with new incoming filters from datasource or configurationAttribute. Let's say there are already filters on the view state. Shouldn't the view state filters then not be updated as well with the new incoming filter?

Reply:
Nope, because just the view state is important here and the view state is retrieved once when the component is mounting.